src/Microsoft.Extensions.Logging.Abstractions/LoggerExtensions.cs

```diff
@@ -425,7 +425,7 @@ public static void Log(this ILogger logger, LogLevel logLevel, EventId eventId,
 
         //------------------------------------------HELPERS------------------------------------------//
 
-        private static string MessageFormatter(object state, Exception error)
+        private static string MessageFormatter(FormattedLogValues state, Exception error)
```

Comment:
`ILogger.Log` takes `Func<TState, Exception, string> formatter` so unfortunately, it won't help.

Reply:
Shame, currently it will make a copy to stack (>8bytes) to get value semantics, then pass the pointer to that stack as the actual argument (due to [ABI](https://github.com/dotnet/coreclr/blob/master/Documentation/botr/clr-abi.md)), then it might make a second copy from that stack at the start of the called method (not sure about this bit)

Reply:
Wait, `Func<T>` is `<in T>`

Reply:
That's variance :(